tests/FakeItEasy.Specs/ArgumentEqualityComparerSpecs.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace FakeItEasy.Specs
+{
+    using FluentAssertions;
+    using Xbehave;
+
+    public static class ArgumentEqualityComparerSpecs
+    {
+        [Scenario]
+        public static void CustomArgumentEqualityComparer(IFoo fake, int result)
+        {
+            "Given a fake"
+                .x(() => fake = A.Fake<IFoo>());
+
+            "And a type for which a custom argument equality comparer exists"
+                .See<ClassWithCustomArgumentEqualityComparer>();
+
+            "When a call to the fake is configured with a specific argument value"
+                .x(() => A.CallTo(() => fake.Bar(new ClassWithCustomArgumentEqualityComparer { Value = 1 })).Returns(42));
+
+            "And a call to the fake is made with a distinct but identical instance"
+                .x(() => result = fake.Bar(new ClassWithCustomArgumentEqualityComparer { Value = 1 }));
+
+            "Then it should return the configured value"
+                .x(() => result.Should().Be(42));
+        }
+
+        [Scenario]
+        public static void TwoCustomArgumentEqualityComparers(IFoo fake, int result)
+        {
+            "Given a fake"
+                .x(() => fake = A.Fake<IFoo>());
+
+            "And a type for which two custom argument equality comparer exist"

```

Comment:
```suggestion
            "And a type for which two custom argument equality comparers exist"
```

and why won't GitHub let me make a code change suggestion when I'm looking at only one of the commits in a PR?